sdk/nodejs/dagger/api/utils.ts

```diff
@@ -0,0 +1,42 @@
+  export function queryBuilder(q: any) {
+
+    const args = (item: any) => { 
+      let regex = /\{"[a-zA-Z]+"/ig;
+      
+      return Object.entries(item.args)
+        .map(value => {
+            if(typeof value[1] === 'object') {
+              return `${value[0]}: ${JSON.stringify(value[1]).replace(regex, str => str.replace(/"/g, ''))}`
+            }
+            if(typeof value[1] === 'number') {
+              return `${value[0]}: ${value[1]}`
+            }
+          return `${value[0]}: "${value[1]}"`
+        })
+    }
+
+    let query = ""
+    q.forEach((item: any, index: any) => {
+      query += `
+        ${item.operation} ${item.args ? `(${args(item)})` : ''} ${q.length - 1 !== index ? '{' : '}'.repeat(q.length - 1)}  
+      `
+    })
+
+    return query.replace(/\s+/g, '')
+  }
+
+  export function queryFlatten(res: any) {
```

Comment:
Just to be sure, this function will transform an array of any depth into a single-dimensional array?

Reply:
This function will return the latest element of a graphQL response. 

graphql response:
```json
 {
    "data": {
      "container": {
        "from": {
          "exec": {
            "stdout": {
              "id": "eyJsbGIiOnsiZGVmIjpbIkdqQUt...."
            }
          }
        }
      }
    }
  }
  ```
with queryFlatten:
```json
{ "id": "eyJsbGIiOnsiZGVmIjpbIkdqQUt...."}
```